STM32Cube/Core/Src/my2c.c

```diff
@@ -0,0 +1,144 @@
+#include <stdbool.h>
+#include "my2c.h"
+
+#include "stm32h7xx_hal.h"
+#include "stm32h7xx_hal_i2c.h"
+
+#include "FreeRTOS.h"
+#include "semphr.h"
+
+SemaphoreHandle_t I2C4BinarySemaphore;
+
+extern I2C_HandleTypeDef hi2c4;
+
+uint8_t timeout_MS = 200;
+
+/**
+ * Register i2c4 callbacks, semaphore
+ */
+bool MY2C_init(void)
+{
+    if (HAL_I2C_RegisterCallback(&hi2c4, HAL_I2C_MEM_TX_COMPLETE_CB_ID, &I2C4_MemTxCallback) != HAL_OK)
+    {
+        return false;
+    }
+
+    if (HAL_I2C_RegisterCallback(&hi2c4, HAL_I2C_MEM_RX_COMPLETE_CB_ID, &I2C4_MemRxCallback) != HAL_OK)
+    {
+        return false;
+    }
+
+    if (HAL_I2C_RegisterCallback(&hi2c4, HAL_I2C_ERROR_CB_ID, I2C4_ErrorCallback) != HAL_OK)
+    {
+        return false;
+    }
+
+	I2C4BinarySemaphore = xSemaphoreCreateBinary();
+
+	if (I2C4BinarySemaphore)
```

Comment:
this should be an explicit comparison, mostly for clarity, see example from the docs

```
    xSemaphore = xSemaphoreCreateBinary();
    if( xSemaphore == NULL )
    {
        /* There was insufficient FreeRTOS heap available for the semaphore to
        be created successfully. */
    }
    else
    {
        /* The semaphore can now be used. Its handle is stored in the
        xSemahore variable.  Calling xSemaphoreTake() on the semaphore here
        will fail until the semaphore has first been given. */
    }```